enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+- As an OCP user/cluster-admin I would like to see the latest Insights status.
+- As an OCP user/cluster-admin I would like to exclude/include particular gatherers.
+
+### API Extensions
+
+This is adding a new `insights.config.openshift.io` API. The proposal of the structure of the new API is mentioned in the [Proposal](#Proposal) section.
+
+### Risks and Mitigations
+
+Risk: Insights operator is not able to read the new config API.
+
+Mitigation: The Insights operator is marked as unvailable through the corresponding clusteroperator condition.
+
+Risk: Somebody/something is repeatedly updating the `ForceGatheringReason` attribute and thus forcing gathering. There is a risk that 
+the cluster would upload data too often, but this assumes that the "repeating interval" allows uploading (if it's forced too quickly then there will be no upload, because the gathering is not over).
+
+Mitigation: Maybe we should introduce some limit on the number of forced gatherings within 1 hour, for example. 
+
+## Design Details
+
+### On demand data gathering options
+
+The on demand data gathering can happen almost immediately when there is no active gathering in the progress, but when there is some in progress 
+then the question is what to do in this situation. There are following options:
+
+- schedule it - e.g using something like the `ScheduleGatheringAt` attribute where the user would provide some time in the future.
+  - Pros 
+    - no need to interrupt gathering in progress
+    - we could probably use the `RateLimiting` queue and add a new gathering to the queue with the `AddAfter` method. Otherwise we would need to define some waiting time (when there’s already some gathering in progress) and right now it’s difficult to estimate an average gathering time since the gatherers run in parallel
+  - Cons
+    - user has to define the scheduling time correctly in the future
+    - gathering doesn’t happen immediately
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [roviding a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+The second option of forcing it is preferred and it also corresponds to the API proposal in the [Proposal](#proposal) section.  
+
+### Open Questions
+
+- ~~Can the API define default values? How does it work or how is it implemented?~~ This is done by using `// +kubebuilder:default=`
+- Should the new time of forced gathering (if any defined by user) set a new time for the existing gathering interval? If I force gathering at time A, will the new automatic gathering be in time A + interval?
```

Comment:
seems reasonable that you'd reset the delay until the next gather, after doing a force gather.

otherwise you risk ending up doing them back to back(one forced, immediately followed by the regularly scheduled one) which doesn't seem useful.

Reply:
Yeah I lean towards this answer as well.

Reply:
Updated.